x/leverage/keeper/inspector.go

```diff
@@ -121,6 +122,65 @@ func (q Querier) Inspect(
 	return &types.QueryInspectResponse{Borrowers: sortedBorrowers}, nil
 }
 
+// Separated from grpc_query.go
+func (q Querier) InspectAccount(
+	goCtx context.Context,
+	req *types.QueryInspectAccount,
+) (*types.QueryInspectAccountResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	k, ctx := q.Keeper, sdk.UnwrapSDKContext(goCtx)
+	tokens := k.GetAllRegisteredTokens(ctx)
+
+	exchangeRates := map[string]tokenExchangeRate{}
+	for _, t := range tokens {
+		exchangeRates[t.BaseDenom] = tokenExchangeRate{
+			symbol:       t.SymbolDenom,
+			exponent:     t.Exponent,
+			exchangeRate: k.DeriveExchangeRate(ctx, t.BaseDenom),
+		}
+	}
+
+	addr, err := sdk.AccAddressFromBech32(req.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	position, err := k.GetAccountPosition(ctx, addr, true)
+	if err != nil {
+		return nil, err
+	}
+	borrowedValue := position.BorrowedValue()
+	collateralValue := position.CollateralValue()
+	liquidationThreshold := position.Limit()
+
+	borrowPosition, err := k.GetAccountPosition(ctx, addr, false)
+	if err != nil {
+		return nil, err
+	}
+
+	borrowed := k.GetBorrowerBorrows(ctx, addr)
+	collateral := k.GetBorrowerCollateral(ctx, addr)
+
+	account := types.InspectAccount{
+		Address: addr.String(),
+		Analysis: &types.RiskInfo{
+			Borrowed:    neat(borrowedValue),
+			Liquidation: neat(liquidationThreshold),
+			Value:       neat(collateralValue),
+		},
+		Position: &types.DecBalances{
+			Collateral: symbolDecCoins(collateral, exchangeRates),
+			Borrowed:   symbolDecCoins(borrowed, exchangeRates),
+		},
+		Info: position.String() + "\n\n" + borrowPosition.String(),
```

Comment:
This string field is meant to be unstable, but also easier to manipulate in patch releases or modified binaries without changing proto.

Currently, when inspecting a single account, it will show the account's positions arranged both by borrow logic and liquidation logic, for a good amount of visibility.

Since this query is single target, it will be available on any node without causing iteration-based DDoS-like problems.